### PR TITLE
Add benchmark scripts for trees

### LINKED ISSUE #315: Add benchmarking script for trees
The scripts will allow the replication of the results in Section 6.1.1 of the paper.

### PR DESCRIPTION
**Pull request by interesaaat**
Closes #315. This should go after #319 and #323.

### CODE REVIEW

**Comment by interesaaat** [comment]
This is fixed!

File: benchmarks/trees/run.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+            "Subset of rows in the datasets to use. Useful for test running "
+            "benchmarks on small amounts of data. WARNING: Some datasets will "
+            "give incorrect accuracy results if nrows is specified as they have "
+            "predefined train/test splits."
+        ),
+    )
+    parser.add_argument("-niters", default=1, type=int, help=("Number of iterations for each experiment"))
+    parser.add_argument("-maxdepth", default=None, type=int, help=("Maxmimum number of levels in the trees"))
+    parser.add_argument(
+        "-validate", default=False, action="store_true", help="Validate prediction output and fails accordigly."
```
Comment (ksaur) [inline benchmarks/trees/run.py]: s/accordigly/accordingly/g

File: doc/html/hummingbird/ml/supported.html [hunk 2]
```diff
@@ -210,8 +210,8 @@ <h1 class="title">Module <code>hummingbird.ml.supported</code></h1>
         from sklearn.decomposition import PCA, KernelPCA, FastICA, TruncatedSVD
 
         # KNeighbors models
-        from sklearn.neighbors._classification import KNeighborsClassifier
-        from sklearn.neighbors._regression import KNeighborsRegressor
+        from sklearn.neighbors import KNeighborsClassifier
```
Comment (ksaur) [inline doc/html/hummingbird/ml/supported.html:213]: Why this one random html?

**Review by ksaur — APPROVED** [review]
I have stared at/thoroughly used this code for like 9 months now so....It's hard to be a fresh set of eyes.

File: benchmarks/trees/run.py (hunk 1 above)
Comment (interesaaat) [reply benchmarks/trees/run.py]: Good catch!

File: doc/html/hummingbird/ml/supported.html (hunk 2 above)
Comment (interesaaat) [reply doc/html/hummingbird/ml/supported.html:213]: ah I actually changed this because it was giving problem with sklearn 0.21.3

**Pull request merged**